src/components/graph/graph.coords.js

```diff
@@ -0,0 +1,333 @@
+ */
+function calcSquareVectorLength(nodeSize, { x, y }, directionVector, isCustomNode) {
+  let edgeSize;
+  if (isCustomNode) {
+    // nodeSize equals the edgeSize in the case of custome-node.
+    edgeSize = nodeSize / 10;
+  } else {
+    // All the edges of a square are equal, inorder to calc it's size we multplie two edges.
+    edgeSize = Math.sqrt(nodeSize);
+  }
+
+  // The x and y coords in this library, represent the top center of the component.
+  const leftSquareX = x - edgeSize / 2;
+  const topSquareY = y - edgeSize / 2;
+
+  return calcRectangleVectorLengthFromCoords(
+    { x1: leftSquareX, y1: topSquareY, x2: leftSquareX + edgeSize, y2: topSquareY + edgeSize },
+    { px: x, py: y },
+    directionVector
+  );
+}
+
+/**
+ * Calculate a the vector length from the center of a rectangle to it's perimeter.
+ *
+ * @param {number} nodeSize The size of the square, when no viewGenerator is specified, else the size of an edge of the viewGenerator square.
+ * @param {Object.<string, number>} nodeCoords The coords of a the square node.
+ * @param {Object.<string, number>} directionVector a 2D vector with x and y components.
+ * @returns {number} The length of the vector from the center of the symbol to it's closet edge, considering the given direction vector.
+ */
+function calcRectangleVectorLength(nodeSize, { x, y }, directionVector) {
+  const horizEdgeSize = nodeSize.width / 10;
+  const vertEdgeSize = nodeSize.height / 10;
+
+  // The x and y coords in this library, represent the top center of the component.
+  const leftSquareX = x - horizEdgeSize / 2;
+  const topSquareY = y - vertEdgeSize / 2;
+
+  // The size between the square center, to the appropriate square edges
+  return calcRectangleVectorLengthFromCoords(
+    { x1: leftSquareX, y1: topSquareY, x2: leftSquareX + horizEdgeSize, y2: topSquareY + vertEdgeSize },
+    { px: x, py: y },
+    directionVector
+  );
+}
+
+/**
+ * Calculate a the vector length of symbol that included in symbols with optimized positioning.
+ *
+ * @param {string} symbolType the string that specifies the symbol type (should be one of {@link #node-symbol-type|node.symbolType})
+ * @param {(number | Object.<string, number>)} nodeSize The size of the square, when no viewGenerator is specified, else the size of an edge of the viewGenerator square.
+ * @param {Object.<string, number>} nodeCoords The coords of a the square node.
+ * @param {Object.<string, number>} directionVector a 2D vector with x and y components.
+ * @param {boolean} isCustomNode is viewGenerator specified.
+ * @returns {number} The length of the vector from the center of the symbol to it's closet edge, considering the given direction vector.
+ */
+function calcVectorLength(symbolType, nodeSize, { x, y }, directionVector, isCustomNode) {
+  if (typeof nodeSize === "object" && nodeSize?.width && nodeSize?.height) {
+    return calcRectangleVectorLength(nodeSize, { x, y }, directionVector);
```

Comment:
This seems strange that we are *hacking* the symbol type whenever a `width` and `height` are provided. What do you think?

Reply:
Size as an `object`, currently only makes sense when a `viewGenerator` is provided.
This issue predates this PR.
Do you think a special `SYMBOL` that only supported with `viewGenerator` is the answer?